redfish-core/lib/account_service.hpp

```diff
@@ -1377,6 +1377,141 @@ inline void updateUserProperties(std::shared_ptr<bmcweb::AsyncResp> asyncResp,
+    {
+        BMCWEB_LOG_ERROR << "Illegal Property ";
+        messages::propertyMissing(asyncResp->res, "IBM");
+        return;
+    }
+
+    if (ibm)
+    {
+        std::optional<nlohmann::json> acf;
+        if (!redfish::json_util::readJson(*ibm, asyncResp->res, "ACF", acf))
+        {
+            BMCWEB_LOG_ERROR << "Illegal Property ";
+            messages::propertyMissing(asyncResp->res, "ACF");
+            return;
+        }
+
+        if (acf)
+        {
+            std::vector<uint8_t> decodedAcf;
+            std::optional<std::string> acfFile;
+            if (acf.value().contains("ACFFile") &&
+                acf.value()["ACFFile"] == nullptr)
+            {
+                acfFile = "";
+            }
+            else
+            {
+                if (!redfish::json_util::readJson(*acf, asyncResp->res,
+                                                  "ACFFile", acfFile))
+                {
+                    BMCWEB_LOG_ERROR << "Illegal Property ";
+                    messages::propertyMissing(asyncResp->res, "ACFFile");
+                    return;
+                }
+
+                std::string sDecodedAcf;
+                if (!crow::utility::base64Decode(*acfFile, sDecodedAcf))
+                {
+                    BMCWEB_LOG_ERROR << "base64 decode failure ";
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+                try
+                {
+                    std::copy(sDecodedAcf.begin(), sDecodedAcf.end(),
+                              std::back_inserter(decodedAcf));
+                }
+                catch (const std::exception& e)
+                {
+                    BMCWEB_LOG_ERROR << e.what();
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+            }
+
+            crow::connections::systemBus->async_method_call(
+                [asyncResp, decodedAcf](const boost::system::error_code ec,
+                                        const std::variant<bool>& retVal) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "Failed to read ACFWindowActive property";
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    const bool* isACFWindowActive = std::get_if<bool>(&retVal);
+
+                    if (isACFWindowActive == nullptr)
+                    {
+                        BMCWEB_LOG_ERROR << "nullptr for ACFWindowActive";
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    if (*isACFWindowActive == true)
+                    {
+                        uploadACF(asyncResp, decodedAcf);
+                        return;
+                    }
+                    else
```

Comment:
clang-tidy wants to collapse the else section (because of the "return" in the  then clause above.  That is, remove the "else {" and closing "}", and unnest the code.

Reply:
okay